### DISCUSSION TITLE
horovod running on cluster with mixed types of machines on k8s

### DISCUSSION THREAD

**Discussion by weidezhang** [horovod/horovod#2793] [Q&A]
Hi, 

I'm have a set up of machines with mostly cpu machines and a few gpu machines and uses k8s to do resource scheduling.  I'm wondering if i can have horovod distributed training to be scheduled on multiple gpu machines only using k8s?

**Comment by tgaddair** [comment]
Hey @weidezhang, yes there are a few good ways to running Horovod on Kubernetes, though I would make sure that you only run Horovod on the GPU machines.

As a starting point, I would try [Horovod on Ray](https://horovod.readthedocs.io/en/stable/ray_include.html) or the Kubeflow [MPI Operator](https://github.com/kubeflow/mpi-operator).
